lib/ExtUtils/MM_Unix.pm

```diff
@@ -2482,12 +2482,12 @@ $(MAKE_APERL_FILE) : static $(FIRST_MAKEFILE) pm_to_blib
 		MAKEAPERL=1 NORECURS=1 CCCDLFLAGS=};
 
 	foreach (@ARGV){
```

Comment:
> A further thought: it will probably best to handle (and also test for) ARGV values that have quotes in...

Agreed. However, I would consider the handling embedded quotes as a separate issue (to be fixed in a separate patch with more tests).  The fix here is specifically to avoid the ARGV pollution which should never happen.

> Now I think of it: foreach my $arg (@ARGV) ...

@Leont hinted at this, but to be clear, if you do this you're still getting an alias to the value within the array.  It's an interesting and incredibly subtle feature of Perl which makes it non-obvious that what you suggest above doesn't work as you're probably expecting.  Feel free to test/try that though to see this in action.